File: npm/src/types/apps.ts
import type { ApiEnvelope, JsonObject, ListOptions, TaskStatus } from './common.js';
import type { GenerationResult } from './generations.js';

export type AppInputParameter = {
  name: string;
  type: 'string' | 'number' | 'boolean' | 'array' | string;
  values?: unknown[];
};

export type PublicApp = {
  id: string;
  input_parameters: AppInputParameter[];
};

export type AppListOptions = ListOptions;

export type PublicAppsEnvelope = ApiEnvelope<PublicApp[]>;
export type PublicAppEnvelope = ApiEnvelope<PublicApp>;

export type CreateAppGenerationRequest = JsonObject;

export type AppGenerationTask = {
  id: string;
  status: TaskStatus;
  result?: GenerationResult;
  error?: {
    code?: string;
    message: string;
  };
};

export type AppGenerationTaskEnvelope = ApiEnvelope<AppGenerationTask>;

File: python/src/buble_ai/types/apps.py
from __future__ import annotations

from typing import Any, List, TypedDict

from .generations import GenerationResult
from .common import TaskStatus


class AppInputParameter(TypedDict, total=False):
    name: str
    type: str
    values: List[Any]


class PublicApp(TypedDict):
    id: str
    input_parameters: List[AppInputParameter]


class PublicAppsEnvelope(TypedDict):
    data: List[PublicApp]


class PublicAppEnvelope(TypedDict):
    data: PublicApp


class AppGenerationTask(TypedDict, total=False):
    id: str
    status: TaskStatus
    result: GenerationResult
    error: dict


class AppGenerationTaskEnvelope(TypedDict):
    data: AppGenerationTask

